### PR TITLE
Prepare Release 1.27.0

### PR DESCRIPTION
**Pull request by cmslewis**
:fireworks: **Highlights**: `Table` __`scrollToRegion(region: IRegion)`__ instance method, `Table` __`renderMode`__ to toggle batched cell rendering, `Table` __`onCompleteRender`__ prop, new `Suggest` component

:book: **Latest docs**: [blueprintjs.com/docs](http://blueprintjs.com/docs)

### 🐛  Bug fixes
- #1498 `pt-file-upload` now adds `ellipsis` styling to long, overflowing file names.
image
- #1510 `Tabs2` now offers a `large` prop to enable `pt-large` styling.
    - ⚠️ This new prop addresses a reported `pt-large` regression from our last release (see: #1506).

### 🐛  Bug fixes
- #1474 `DateTimePicker` no longer clears selected date when changing time values. (🎩 @cathyxz)

### 💎  New features
- #1496 🌟 `Table` now offers a __`scrollToRegion(region: IRegion)`__ instance method to programmatically scroll a desired row, column, or cell into view!
    - The target region will be positioned in the top/left corner of the viewport.
        - Or if the table is maximally scrolled, then having the target cell simply in view is enough.
    - The target region will account for any frozen rows and/or columns and adjust its position accordingly to stay visible.
    - The scrolling happens instantaneously for now; in the future, we hope to add an `animated?: boolean` option to animate the scroll (see: #1501).
- #1503 🌟  `Table` now supports a __`renderMode?: RenderMode`__ prop:
    - `RenderMode.BATCH` _(default)_: renders cells in batches to improve performance (see: #1201)
    - `RenderMode.NONE`: renders cells synchronously all at once
- #1505 🌟  `Table` now offers an __`onCompleteRender`__ prop.
  - This callback is invoked on mount/update when all cells in view have finished rendering.
  - This is particularly useful when batched cell rendering is enabled.

### 🐛  Bug fixes

- #1447 `Table` now shows the focused cell immediately and in the right place when `enableFocus` becomes `true`.
- #1502 `Table` resizing columns no longer throws error when `isRowHeaderShown=false`
- #1505 `Table` no longer fires `onVisibleCellsChange` redundantly when the viewport hasn't actually changed.
- #1507 `Table` `FULL_ROWS`, `FULL_COLUMNS`, and `FULL_TABLE` selection overlays now align properly to cell borders.

### 💎  New features
- #1499 🌟 New **`Suggest`** component!
    - `Suggest` behaves similarly to `Select`, except it renders a text input as the `Popover` target instead of arbitrary children.

### CODE REVIEW

**Comment by tgreenwatts** [comment]
minor thoughts imo, looks good

Need to fill in the version numbers
Tabs feels like a bugfix or enhancement to me, not new feature?
Should we add a note to Scroll to region that animations will follow?
Nit: the 1201 reference for onComplete doesn't seem necessary
"perfectly" - you're just asking for someone to prove you wrong

File: packages/docs/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "style": "dist/docs.css",
   "unpkg": "dist/docs.bundle.js",
   "dependencies": {
-    "@blueprintjs/core": "^1.25.1",
+    "@blueprintjs/core": "^1.27.0",
```
Comment (giladgray) [inline packages/docs/package.json]: revert

File: packages/site-landing/package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "watch": "onchange 'src/**' -- npm-run-all build:copy build:webpack"
   },
   "devDependencies": {
-    "@blueprintjs/core": "^1.25.1",
+    "@blueprintjs/core": "^1.27.0",
```
Comment (giladgray) [inline packages/site-landing/package.json]: revert

File: packages/docs/package.json (hunk 1 above)
Comment (cmslewis) [reply packages/docs/package.json]: What's the rule of thumb with package versions in `site-landing`, `site-docs`, and `docs`? (For both dependencies and the package itself.)

**Comment by cmslewis** [comment]
@giladgray done

**Comment by cmslewis** [comment]
@giladgray @tgreenwatts @gscshoyru I added @cathyxz's #1474 PR to this release. take a look at that and anything else. 👍

**Comment by cmslewis** [comment]
:shipit:

**Pull request merged**